src/Http/Middleware/CspMiddleware.php

```diff
@@ -52,7 +52,7 @@ public function __construct($csp = null)
         }
 
         if (!empty($cspConfig) && is_array($cspConfig)) {
```

Comment:
Why do we need is_array()? If the payload is here, wouldnt the CSPBuilder class throw exception for the wrong type?

Reply:
I see, you are later using the object.

Reply:
Would passing an empty array to the builder even be useful? If not I would throw an exception if `if (empty($cspConfig))` succeeds.

Reply:
I can also make it as simple as:
```php
    public function __construct($csp = null)
    {
        if ($csp === null) {
            $csp = (array)Configure::read('App.CSP');
        }

        if (!$csp instanceof CSPBuilder) {
            $csp = new CSPBuilder($csp);
        }

        $this->csp = $csp;
    }
```

Reply:
No use of `Configure::read()` please. Allow passing array or `CSPBuilder()` as argument, without the default `null`.